aws/src/test/java/org/apache/iceberg/aws/s3/TestS3OutputStream.java

```diff
@@ -119,24 +120,28 @@ public void testWrite() {
 
   @Test
   public void testAbortAfterFailedPartUpload() {
-    doThrow(new RuntimeException()).when(s3mock).uploadPart((UploadPartRequest) any(), (RequestBody) any());
+    doThrow(new RuntimeException("mock uploadPart failure"))
+            .when(s3mock).uploadPart((UploadPartRequest) any(), (RequestBody) any());
 
-    try (S3OutputStream stream = new S3OutputStream(s3mock, randomURI(), properties, nullMetrics())) {
-      stream.write(randomData(10 * 1024 * 1024));
-    } catch (Exception e) {
-      verify(s3mock, atLeastOnce()).abortMultipartUpload((AbortMultipartUploadRequest) any());
-    }
+    Assertions.assertThrows(RuntimeException.class, ()->{
+      try (S3OutputStream stream = new S3OutputStream(s3mock, randomURI(), properties, nullMetrics())) {
+        stream.write(randomData(10 * 1024 * 1024));
+      }
+    });
+    verify(s3mock, atLeastOnce()).abortMultipartUpload((AbortMultipartUploadRequest) any());
   }
 
   @Test
   public void testAbortMultipart() {
-    doThrow(new RuntimeException()).when(s3mock).completeMultipartUpload((CompleteMultipartUploadRequest) any());
+    doThrow(new RuntimeException("mock completeMultipartUpload failure"))
+            .when(s3mock).completeMultipartUpload((CompleteMultipartUploadRequest) any());
 
-    try (S3OutputStream stream = new S3OutputStream(s3mock, randomURI(), properties, nullMetrics())) {
-      stream.write(randomData(10 * 1024 * 1024));
-    } catch (Exception e) {
-      verify(s3mock).abortMultipartUpload((AbortMultipartUploadRequest) any());
-    }
+    Assertions.assertThrows(RuntimeException.class, ()->{
+      try (S3OutputStream stream = new S3OutputStream(s3mock, randomURI(), properties, nullMetrics())) {
+        stream.write(randomData(10 * 1024 * 1024));
+      }
+    });
```

Comment:
We should probably add checking the message since that was added as well, Also please be sure to get that checkstyle running ```gradle build -x test  -x integrationTest``` should run all the static checks.